zachsinterview/src/main/java/com/imprev/zachsinterview/NumberStoreAdditionalOperations.java

```diff
@@ -0,0 +1,13 @@
+package com.imprev.zachsinterview;
+
+/**
+ * Provides some additional methods we may want to extend our NumericalKeyValueStore library with
+ */
+public interface NumberStoreAdditionalOperations {
```

Comment:
A couple pieces of feedback on this part:

(a) a more descriptive name could help make identifying the purpose of this interface more clear
(b) inheriting from the base interface NumStore would more clearly cement the relationship between the two interfaces and enforce that this interface implements all needed methods